core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushLimitThroughOuterJoin.java

```diff
@@ -57,6 +58,7 @@
 
     private static final Pattern<LimitNode> PATTERN = limit()
             .matching(limit -> !limit.isWithTies())
+            .with(isOrderSensitive().equalTo(false))
```

Comment:
If the ordering is dependent only on the columns from the left side of a LEFT join or the right side of a RIGHT join, it should be possible to push down. Join doesn't change the relative order of the input